libs/utils/src/Panic.cpp

```diff
@@ -139,6 +152,11 @@ void TPanic<T>::panic(char const* function, char const* file, int line, const ch
     va_end(args);
     T e(function, formatFile(file), line, reason);
     e.log();
+
+#if defined(__APPLE__)
+    details::throwDarwinException(e);
```

Comment:
Actually, we may want to give users the option of which type of exception they want thrown, maybe through a compile-time flag. @pixelflinger thoughts?

Reply:
I'm thinking we should throw the C++ exception first if enabled. If not it'll revert to the darwinException.
If the user wants a darwin exception that can always throw one from their C++ exception handler.